### ISSUE TITLE
Data in editable children lost without warning when moving node in tree

### ISSUE THREAD

**Issue by lekoder** [godotengine/godot#23979]
**Godot version:**
3.1.aplha. c025f52

**OS/device including version:**
Windows 10

**Issue description:**
When moving an instanced scene with editable children, the "editable children" flag is lost. Moving the scene back re-enables the flag. If scene is saved when instanced scene is in new location, all children data is lost without warning.

**Expected behaviour:**
When moving an instanced scene with editable children, all modifications are preserved.

**Actual behaviour:**
When moving an instanced scene with editable children, editable flag is lost. Child modification data is lost as soon as scene is saved to disk.

**Steps to reproduce:**
1. Instance a scene with editable children.
1. Make modification to the children. 
1. Move the instanced scene in scene tree. 
1. Save and reopen.

Probably related to #20409 and #6455.

**Comment by KoBeWi** [comment]
Still valid in 0cd98ec

**Comment by chucklepie** [comment]
Is this the same issue:

1. I added an inherited scene
2. I made a ton of modifications
3. I renamed the node
4. It reverted back to non-inherited and I lost every single change I made :(

i.e. I didn't even move, duplicate or instance anything. I simply renamed a node after editing it.

I've done this so many times now it's really annoying me and wonder why it hasn't been fixed yet as it seems pretty nasty and horrible, especially given you won't be saving or committing files during the process of simply renaming a node....
